ref-exchange/src/account_deposit.rs

```diff
@@ -173,30 +243,43 @@ impl Contract {
                         )
                         .as_bytes(),
                     );
-                    let mut owner_account = self.get_account_deposits(&self.owner_id);
+                    let mut owner_account = self.internal_unwrap_account(&self.owner_id);
                     owner_account.deposit(&token_id, amount.0);
-                    self.internal_save_account(&self.owner_id.clone(), owner_account);
+                    // owner account is out of storage management!!!
+                    self.internal_save_account(&self.owner_id.clone(), owner_account, 0);
                 }
             }
         };
     }
 }
 
 impl Contract {
+
     /// Checks that account has enough storage to be stored and saves it into collection.
     /// This should be only place to directly use `self.accounts`.
-    pub(crate) fn internal_save_account(&mut self, account_id: &AccountId, account: Account) {
-        account.assert_storage_usage();
-        self.accounts.insert(&account_id, &account);
+    /// [FEATURE_STORAGE_USED]
+    /// Param `prev` set to zero means caller ensure the `storage_used` won't change.
+    pub(crate) fn internal_save_account(&mut self, account_id: &AccountId, mut account: Account, prev: StorageUsage) {
+        // account.assert_storage_usage();
+        if prev > 0 {
```

Comment:
in general, I would expect that storage of the account is asseted when account is changed, not when it's saved